www/include/TestResults.php

```diff
@@ -186,8 +186,17 @@ public function getMedianRunNumber($metric, $cached, $medianMode = "median") {
     if ($numValues == 1 || $medianMode == "fastest") {
       return $runNumbers[0];
     }
-    $medianIndex = (int)floor($numValues / 2.0);
-    return $runNumbers[$medianIndex];
+    $medianIndex = (int)floor(((float)$numValues + 1.0) / 2.0);
```

Comment:
Add a ` - 1` and we go from a 1-based index to a 0-based index. Then we can still simply return `$runNumbers[$medianIndex]` and drop the loop. The loop below does nothing else than choosing the value by using $current as a 1-based index and comparing it with $medianIndex (also $time is misleading, as the array contains run numbers).

With `(int)floor(((float)$numValues + 1.0) / 2.0) -1`:
* $numValues = 2  →  0
* $numValues = 3  →  1
* $numValues = 4  →  1
* $numValues = 5  →  2
* ...

So it's always the lower 0-based median index.